volatility3/framework/plugins/windows/amcache.py

```diff
@@ -0,0 +1,646 @@
+                    sha1_hash.lstrip("0000")
+                    if isinstance(sha1_hash, str)
+                    else sha1_hash
+                ),
+                product_name=product_name,
+            )
+
+    @classmethod
+    def parse_programs_key(
+        cls, programs_key: reg_extensions.CM_KEY_NODE
+    ) -> Iterator[Tuple[str, _AmcacheEntry]]:
+        """Parses Program entries from the Windows 8 `Root\\Programs` key.
+
+        :param programs_key: The `Root\\Programs` registry key.
+
+        :return: An iterator of tuples, where the first member is the program ID string for
+        correlating `Root\\File` entries, and the second member is the `AmcacheEntry`.
+        """
+        for program_key in programs_key.get_subkeys():
+            values = {
+                str(value.get_name()): value
+                for value in program_key.get_values()
+                if value.get_name() in [key.value for key in Win8ProgramValName]
+            }
+            vollog.debug(f"Parsing Win8 Program key {program_key.get_name()}")
+            program_id = program_key.get_name().strip().strip("\u0000")
+
+            product = _get_string_value(values, Win8ProgramValName.Product.value)
+            company = _get_string_value(values, Win8ProgramValName.Publisher.value)
+            install_time = _get_datetime_utc_epoch_value(
+                values, Win8ProgramValName.InstallTime.value
+            )
+            _version = _get_string_value(values, Win8ProgramValName.Version.value)
+
+            if isinstance(_version, str):
+                if isinstance(product, str):
+                    product = f"{product} {_version}"
+                else:
+                    product = f"UNKNOWN {_version}"
+
+            yield program_id, _AmcacheEntry(
+                AmcacheEntryType.Program.name,
+                company=company,
+                last_modify_time=conversion.wintime_to_datetime(
+                    program_key.LastWriteTime.QuadPart
+                ),
+                install_time=install_time,
+                product_name=product,
+            )
+
+    @classmethod
+    def parse_inventory_app_key(
+        cls, inv_app_key: reg_extensions.CM_KEY_NODE
+    ) -> Iterator[Tuple[str, _AmcacheEntry]]:
+        """Parses InventoryApplication entries from the Windows 10 `Root\\InventoryApplication` key.
+
+        :param programs_key: The `Root\\InventoryApplication` registry key.
+
+        :return: An iterator of tuples, where the first member is the program ID string for
+        correlating `Root\\InventoryApplicationFile` entries, and the second member is the `AmcacheEntry`.
+        """
+        for program_key in inv_app_key.get_subkeys():
+            program_id = program_key.get_name()
+
+            values = {
+                str(value.get_name()): value
+                for value in program_key.get_values()
+                if value.get_name() in [key.value for key in Win10InvAppValName]
```

Comment:
Almost certainly a small point, but this list comprehension looks like it could be done once outside the loop and then just referenced in the loop?